### PR TITLE
feat: Update Honeycomb logger to use EMAThroughput sampler

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?

Updates the Honeycomb logger to use the [EMAThroughput](https://github.com/honeycombio/dynsampler-go/blob/main/emathroughput.go#L77) sampler. This sampler has built-in support for burst protection that the [PerKeyThroughput](https://github.com/honeycombio/dynsampler-go/blob/main/perkeythroughput.go#L17) sampler does not.

Burst protection would be useful when a cluster node does down unexpectedly because the other nodes in the cluster will fail to make peer requests until it is removed from the cluster. This can result in a very high number of "failed to send" log messages in a very small window, faster than what the PerKeyThroughput sampler can adjust the sample rate which results in all log messages being sent. The burst protection from the EMAThroughput sampler would help here as it will schedule an update to sample rates if a high number of events are received very quickly, allowing it to react quicker.

This is a like for like replacement, and such I haven't added any additional configuration options the new sampler supports. We can add more later if desired.

## Short description of the changes
- Replace the Honeycomb logger sampler with the EMAThroughput sampler.

### CODE REVIEW

**Comment by cartermp** [comment]
Will this also aid https://github.com/honeycombio/refinery/issues/1222?

**Comment by MikeGoldsmith** [comment]
> Will this also aid #1222?

Yes, it probably 👍🏻

File: logger/honeycomb.go [hunk 1]
```diff
@@ -57,10 +57,10 @@ func (h *HoneycombLogger) Start() error {
 	}
 
 	if loggerConfig.GetSamplerEnabled() {
-		h.sampler = &dynsampler.PerKeyThroughput{
-			ClearFrequencyDuration: 10 * time.Second,
-			PerKeyThroughputPerSec: loggerConfig.SamplerThroughput,
-			MaxKeys:                1000,
+		h.sampler = &dynsampler.EMAThroughput{
+			AdjustmentInterval:   10 * time.Second,
```
Comment (kentquirk) [inline logger/honeycomb.go:61]: Let's lengthen this interval -- maybe to 30s, please?

**Review by kentquirk — APPROVED** [review]
Sounds good, but given what we've been learning about EMA, we should extend the interval.

**Review by VinozzZ — COMMENTED** [review]
Given all the quirks that we recently discovered in EMA samplers, I'm worried that it's going to be harder to reason why a log line doesn't get sampled

**Comment by MikeGoldsmith** [comment]
We already sample refinery log messages; this change is just give burst protection when a sudden increase increase happens. The throughput samplers ensure any unique log message is sent at least once per window.

**Pull request merged**